qiskit_quantuminspire/qi_jobs.py

```diff
@@ -35,15 +59,78 @@ def __init__(
         super().__init__(backend, job_id, **kwargs)
         self._job_ids: List[str] = []
         self._run_input = run_input
+        self.program_name = None
 
-    def submit(self) -> None:
+    async def submit(self) -> None:
         """Submit the (batch)job to the quantum inspire backend.
 
         Use compute-api-client to call the cjm endpoints in the correct order, to submit the jobs.
         """
-        for i in range(1, 3):
-            self._job_ids.append(str(i))
-        self.job_id = "999"  # ID of the submitted batch-job
+        options = self.backend().options
+        # call create algorithm
+        async with ApiClient(config()) as api_client:
+            project = await self._create_project(api_client)
+            algorithm = await self._create_algorithm(api_client, project)
+            commit = await self._create_commit(api_client, algorithm)
+            file = await self._create_file(api_client, commit)
+            batch_job = await self._create_batch_job(api_client, backend_type_id=self.backend().id)
+            job: Job = await self._create_job(
+                api_client, file, batch_job, number_of_shots=options.get("shots", default=1000)
+            )
+            await self._enqueue_batch_job(api_client, batch_job)
+            self.job_id = job.id
```

Comment:
The QIJob object maps to a batch_job, not to a job.

Reply:
Corrected.